### PR TITLE
Add complex class

### PR DESCRIPTION
**Pull request by iconnor**

### CODE REVIEW

File: src/complex.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
+class Complex
```
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
```
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb (hunk 1 above)
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  def terrible_idea(param)
+    Model.find_by_sql("SELECT #{param}")
+  end
```
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb (hunk 2 above)
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb (hunk 3 above)
Comment (iconnor) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb (hunk 1 above)
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb (hunk 2 above)
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb (hunk 3 above)
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb (hunk 1 above)
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb (hunk 2 above)
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+  private
+
+  def get_cost_per_line(lang, cost_type = :all)
```
Comment (jacepatel) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb (hunk 1 above)
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb (hunk 2 above)
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb (hunk 4 above)
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
+class Complex
```
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
```
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  private
+
+  def get_cost_per_line(lang, cost_type = :all)
+    if lang == 'SUM'
+      0
+    elsif lang =~ /HTML/i
+      front_end(cost_type) ? html : 0
```
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

File: src/complex.rb (hunk 5 above)
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Missing top-level class documentation comment. |

File: src/complex.rb (hunk 6 above)
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 0.5 | Extra empty line detected at class body beginning. |

File: src/complex.rb [hunk 8]
```diff
@@ -0,0 +1,74 @@
+class Complex
+
+  private
+
+  def get_cost_per_line(lang, cost_type = :all)
```
Comment (mikesive) [inline src/complex.rb]: Bliss.AI: report

| Debt | Issue |
|---:|---|
| 24.5 | Cyclomatic complexity for get_cost_per_line is too high. [51/6] |
| 24.5 | Perceived complexity for get_cost_per_line is too high. [52/7] |
| 7.4 | Assignment Branch Condition size for get_cost_per_line is too high. [89.31/15] |
| 4.5 | Method has too many lines. [55/10] |

**Pull request closed without merging**